api/controller/controller_test.go

```diff
@@ -271,3 +273,80 @@ func makeSpec() controller.MigrationSpec {
 func randomUUID() string {
 	return utils.MustNewUUID().String()
 }
+
+func (s *Suite) TestModelStatusEmpty(c *gc.C) {
+	apiCaller := apitesting.APICallerFunc(func(objType string, version int, id, request string, arg, result interface{}) error {
+		c.Check(objType, gc.Equals, "Controller")
+		c.Check(id, gc.Equals, "")
+		c.Check(request, gc.Equals, "ModelStatus")
+		c.Check(result, gc.FitsTypeOf, &params.ModelStatusResults{})
+
+		return nil
+	})
+
+	client := controller.NewClient(apiCaller)
+	results, err := client.ModelStatus()
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(results, jc.DeepEquals, []base.ModelStatus{})
+}
+
+func (s *Suite) TestModelStatus(c *gc.C) {
+	apiCaller := apitesting.BestVersionCaller{
+		BestVersion: 4,
```

Comment:
Should we have a BestVersion: 3 test that checks that we get the Error back as the global error instead of per-model errors?

Reply:
This is kind of done in a TestModelStatusError, especially since we are not treating older facades differently now in api layer.
In other words, older facades will return a global error if any of the models fail.